scripts/oecd/regional_demography/deaths/preprocess_csv.py

```diff
@@ -13,120 +13,117 @@
 # limitations under the License.
 
 import sys
-
-sys.path.append('../')
-from utils import multi_index_to_single_index
+import os
 import csv
 import json
 import pandas as pd
+import logging
+from absl import flags
+from absl import app
+from columns import *
+
+_FLAGS = flags.FLAGS
+_MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
+flags.DEFINE_string('mode', '', 'Options: download or process')
+
+logging.basicConfig(level=logging.INFO)
+
+_MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+sys.path.insert(1, os.path.join(_MODULE_DIR, '../'))
+from utils import multi_index_to_single_index
+from download import download_data_to_file_and_df
+
+
+def process_data(df, output_file_path):
+    logging.info("Processing the input file")
+    logging.info("Filtering the required columns")
+    df1 = df[(df['TERRITORIAL_LEVEL'] == "TL3")]
+    df1 = df[[
+        'TERRITORIAL_LEVEL', 'REF_AREA', 'Reference area', 'AGE', 'SEX',
+        'TIME_PERIOD', 'OBS_VALUE', 'UNIT_MEASURE'
+    ]]
+    df1["AGE"] = df1["UNIT_MEASURE"] + "_" + df1["AGE"] + df1["SEX"]
+
+    # First remove geos with names that we don't have mappings to dcid for.
+    try:
+        regid_file = "scripts/oecd/regional_demography/regid2dcid.json"
+        with open(regid_file, 'r') as f:
+            regid2dcid = dict(json.loads(f.read()))
+        logging.info("Resolving places")
+        df1 = df1[df1['REF_AREA'].isin(regid2dcid.keys())]
+        # Second, replace the names with dcids
+        df1['Reference area'] = df1.apply(
+            lambda row: regid2dcid[row['REF_AREA']], axis=1)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        logging.error(f"Error processing regid2dcid.json: {e}")
```

Comment:
can we use logging.fatal here if regid2dcid.json file is missing?

Reply:
using logging.fatal